[Bug]: Update flame to 1.18.0

### What happened?

```
../../../.pub-cache/hosted/pub.dev/flame-1.17.0/lib/src/game/notifying_vector2.dart:189:30: Error: The method 'UnmodifiableFloat64ListView' isn't defined for the class 'NotifyingVector2'.
 - 'NotifyingVector2' is from 'package:flame/src/game/notifying_vector2.dart' ('../../../.pub-cache/hosted/pub.dev/flame-1.17.0/lib/src/game/notifying_vector2.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableFloat64ListView'.
  Float64List get storage => UnmodifiableFloat64ListView(super.storage);
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot_program failed: Exception

Command PhaseScriptExecution failed with a nonzero exit code
warning: FaFaRunnerMac isn't code signed but requires entitlements. It is not possible to add entitlements to a binary without signing it. (in target 'FaFaRunnerMac' from project 'FaFaRunnerMac')
** BUILD FAILED **

Error: Build process failed
```

### Steps to reproduce?

```Markdown
dart 3.5.0
flutter 3.24.0
```


### What did you expect to happen?

The app should run normally.

### Bonfire version

3.9.8

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,6 @@
 # next
+- BREAKING CHANGE: Bump Flutter SDK minimum version to 3.22.0
+- Upgrade: Packages (flame v1.18.0, http v1.2.2, a_star_algorithm v0.3.2).
 - Adds new SceneActions
 
 # 3.9.9

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -7,17 +7,17 @@ issue_tracker: https://github.com/RafaelBarbosatec/bonfire/issues
 # documentation: https://bonfire-engine.github.io
 
 environment:
-  sdk: ">=3.2.0 <4.0.0"
-  flutter: ">=3.13.0"
+  sdk: ">=3.4.0 <4.0.0"
+  flutter: ">=3.22.0"
 
 dependencies:
   flutter:
     sdk: flutter
 
-  flame: 1.17.0
+  flame: 1.18.0
   tiledjsonreader: ^1.3.5
-  http: ^1.1.0
-  a_star_algorithm: ^0.3.1
+  http: ^1.2.2
+  a_star_algorithm: ^0.3.2
 
 dev_dependencies:
   flutter_test:

